django/forms/models.py

```diff
@@ -279,6 +279,13 @@ def __new__(mcs, name, bases, attrs):
                 # fields from the model"
                 opts.fields = None
 
+            if opts.exclude:
```

Comment:
Shouldn't we at least raise an exception when a modelform is instantiated
like that ? Feel free to remove my comments if you think this has security
implications and we can talk this in private. Really I'm not sure at all at
this point.
Le 4 juin 2015 18:58, "James Pic" <jamespic@gmail.com> a écrit :

> Thanks for your advice, I'm wondering, what are the security implications
> of this ? After all, aren't we requiring setting fields or exclude for
> security ?
> Le 4 juin 2015 18:43, "David Gouldin" <notifications@github.com> a écrit :
>
>> In django/forms/models.py
>> <https://github.com/django/django/pull/4747#discussion_r31745049>:
>>
>> > @@ -279,6 +279,13 @@ def __new__(mcs, name, bases, attrs):
>> >                  # fields from the model"
>> >                  opts.fields = None
>> >
>> > +            if opts.exclude:
>>
>> Alternatively, you could handle this where fields is updated with new_class.declared
>> fields like so:
>>
>> for field_name, field in new_class.declared_fields.iteritems():
>>     if opts.exclude and field_name in opts.exclude:
>>         continue
>>     if opts.fields and field_name not in opts.fields:
>>         continue
>>     fields[field_name] = field
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/django/django/pull/4747/files#r31745049>.
>>
>